### DISCUSSION TITLE
Workflow re-run and debugging

### DISCUSSION THREAD

**Discussion by asudhak** [community/community#29868] [Actions]
I'm working on a github workflow that runs `actions/github-script` on a `pull_request_target`. When developing, I obviously need to make some changes and run the workflow to ensure it works as expected. Is there an easy way to run a new version of the workflow on an existing pull request rather than manually creating pull requests in order to trigger the workflow ?

**Comment by philip-gai** [comment]
I would suggest looking at the docs for [Manually running a workflow](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow). Specifically, you can add the `workflow_dispatch` event trigger which will allow you to then manually run the workflow via the UI, using any branch.

IIRC you may have to add the `workflow_dispatch` event trigger to the workflow in your `main` branch before you will see the option of manually running it via the UI.[^1]

[^1]: This suggestion was made on my own, not as an employee of GitHub 🙂
